frontend/src/pages/MessagingQueues/MQDetails/DropRateView/DropRateView.tsx

```diff
@@ -0,0 +1,251 @@
+				return (
+					<Typography.Link
+						onClick={(e): void => {
+							e.preventDefault();
+							e.stopPropagation();
+							window.open(`/services/${encodeURIComponent(text)}`, '_blank');
+						}}
+					>
+						{text}
+					</Typography.Link>
+				);
+			}
+
+			if (column === 'breach_percentage' && text) {
+				if (!isNumber(text))
+					return <Typography.Text>{text.toString()}</Typography.Text>;
+				return (
+					<Typography.Text>
+						{(typeof text === 'string' ? parseFloat(text) : text).toFixed(2)} %
+					</Typography.Text>
+				);
+			}
+
+			return <Typography.Text>{text}</Typography.Text>;
+		},
+	}));
+
+	return columns;
+}
+
+const showPaginationItem = (total: number, range: number[]): JSX.Element => (
+	<>
+		<Typography.Text className="numbers">
+			{range[0]} &#8212; {range[1]}
+		</Typography.Text>
+		<Typography.Text className="total"> of {total}</Typography.Text>
+	</>
+);
+
+function DropRateView(): JSX.Element {
+	const [columns, setColumns] = useState<any[]>([]);
+	const [tableData, setTableData] = useState<any[]>([]);
+	const { notifications } = useNotifications();
+	const { maxTime, minTime } = useSelector<AppState, GlobalReducer>(
+		(state) => state.globalTime,
+	);
+	const [data, setData] = useState<
+		DropRateAPIResponse['data']['result'][0]['list']
+	>([]);
+	const [interval, setInterval] = useState<string>('');
+
+	const [visibleCounts, setVisibleCounts] = useState<Record<number, number>>({});
+
+	const paginationConfig = useMemo(
+		() =>
+			tableData?.length > 10 && {
+				pageSize: 10,
+				showTotal: showPaginationItem,
+				showSizeChanger: false,
+				hideOnSinglePage: true,
+			},
+		[tableData],
+	);
+
+	const evaluationTime = useMemo(() => convertToMilliseconds(interval), [
+		interval,
+	]);
+	const tableApiPayload: MessagingQueueServicePayload = useMemo(
+		() => ({
+			start: minTime,
+			end: maxTime,
+			evalTime: evaluationTime * 1e6,
+		}),
+		[evaluationTime, maxTime, minTime],
+	);
+
+	const handleOnError = (error: Error): void => {
+		notifications.error({
+			message: axios.isAxiosError(error) ? error?.message : SOMETHING_WENT_WRONG,
+		});
+	};
+
+	const handleShowMore = (index: number): void => {
+		setVisibleCounts((prevCounts) => ({
+			...prevCounts,
+			[index]: (prevCounts[index] || 4) + 4,
+		}));
+	};
+
+	const { mutate: getViewDetails, isLoading } = useMutation(getKafkaSpanEval, {
+		onSuccess: (data) => {
+			if (data.payload) {
+				setData(data.payload.result[0].list);
+			}
+		},
+		onError: handleOnError,
+	});
+
+	useEffect(() => {
+		if (data?.length > 0) {
+			setColumns(getColumns(data, visibleCounts, handleShowMore));
+			setTableData(getTableData(data));
+		}
+		// eslint-disable-next-line react-hooks/exhaustive-deps
+	}, [data, visibleCounts]);
+
+	useEffect(() => {
+		if (evaluationTime) {
+			getViewDetails(tableApiPayload);
+		}
+		// eslint-disable-next-line react-hooks/exhaustive-deps
+	}, [minTime, maxTime, evaluationTime]);
```

Comment:
if we use useQuery, we won't need this useEffect. instead we can add the dependencies in the query key of useQuery.